app/controllers/application_controller.rb

```diff
@@ -15,12 +15,18 @@ def return_current_user_projects
     end
   end
 
+  def generate_thumbnail(path, imagefile, commit_id)
+    image = Magick::Image.read("#{path}/satellite/#{imagefile.original_filename}").first
+    image.scale(50,50).write "#{path}/thumbnails/#{commit_id}"
+  end
+
```

Comment:
I think these methods should be present in projects_helper instead of being in application_controller. They are only used in the projects_controller as far as I can tell.